### DISCUSSION TITLE
opret committment and CLI tools

### DISCUSSION THREAD

**Discussion by nicbus** [LNP-BP/nodes#9] [Support]
As far as I can see, there is currently no way to add an `opret` committment using CLI tools.

The [demo script](https://github.com/LNP-BP/nodes/blob/master/contrib/demo-rgb.sh) prepares the PSBT with `btc-cold` and prepares it for a `tapret` committment using the option `--allow-tapret-path` but I see no equivalent for `opret`.

Moreover, the use of `btc-cold construct` implies the PSBT needs to be created using a wallet managed with `btc-hot` / `btc-cold`, while I see the use case for handling PSBTs with other walleting tools and then adding the structure for `tapret` / `opret` committments using RGB-specific tools.

Has this already been thought of?
Is the addition of a dedicated CLI (sub)command to do this already planned?

If it helps, I can volunteer to implement this, but wanted to ask for guidance before I start so there's no duplicate work and to avoid going off in the wrong direction while some thought has already been put into it.

**Comment by cryptoquick** [comment]
This may be related to this issue I filed:
https://github.com/RGB-WG/rust-rgb20/issues/6

**Reply by dr-orlovsky** [reply]
Correct, a fix is needed. Will work on it (feel free to propose a PR since it will take few more days on my side to get to it)

**Comment by oacdutra** [comment]
Hi @nicbus,

I started work on improviments to allow use opret1st with LNPBP libs. See details here:

- https://github.com/RGB-WG/rgb-std/pull/8
- https://github.com/RGB-WG/rust-rgb20/pull/10

To complete the task, we need to update `btc-cold` to create the opret1st correctly when we do not use the `--allow-tapret-path` parameter. 

I think the good way to do this involves some steps:
1. The psbt file needs two outputs: one for bitcoin change address   and another to commitment.
2. The psbt output for commitment will need to be marked as **opret_host**.

After that, we need split the `rgb-demo.sh` by commitment scheme.

PS: I'm working on others improviments, but I plan to return to that soon.
